From 37e34e02db1d576753c9119e26d298b04de5b711 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:28:10 -0400
Subject: [infiniband] ipoib: Remove unnecessary test for NULL before debugfs_remove()

Message-id: <608c489be73b96054d1fdac49cb1448d591fa992.1411578494.git.dledford@redhat.com>
Patchwork-id: 96671
O-Subject: [Patch RHEL7 17/26] IPoIB: Remove unnecessary test for NULL before debugfs_remove()
Bugzilla: 1107621
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Fabian Frederick <fabf@skynet.be>

Fix checkpatch warning:

    WARNING: debugfs_remove(NULL) is safe this check is probably not required

Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit e42fa2092c1049ac9c0e38aaac39ef3c40e91a36)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/ipoib/ipoib_fs.c b/drivers/infiniband/ulp/ipoib/ipoib_fs.c
index 5006185..6bd5740 100644
--- a/drivers/infiniband/ulp/ipoib/ipoib_fs.c
+++ b/drivers/infiniband/ulp/ipoib/ipoib_fs.c
@@ -281,10 +281,8 @@ void ipoib_delete_debug_files(struct net_device *dev)
 {
  struct ipoib_dev_priv *priv = netdev_priv(dev);
 
- if (priv->mcg_dentry)
-  debugfs_remove(priv->mcg_dentry);
- if (priv->path_dentry)
-  debugfs_remove(priv->path_dentry);
+ debugfs_remove(priv->mcg_dentry);
+ debugfs_remove(priv->path_dentry);
 }
 
 int ipoib_register_debugfs(void)
-- 
1.7.1